src/components/Text/Text.module.css

```diff
@@ -1,12 +1,101 @@
 @import '../../design-tokens/mixins.css';
 
 /*------------------------------------*\
-    #TEXT
+    #Text
 \*------------------------------------*/
 
 /**
- * 1) 
+ * 1) h1, h2, h3, h4, h5, h6
+ * 2) Styles are either controlled by as or size declaration
  */
-/* stylelint-disable-next-line block-no-empty */
-.text {
+.text,
+.text--body {
+  @mixin reset;
+  
+  @mixin eds-theme-typography-body-text;
+}
+
+/**
+ * Sizes
+ */
+.text--h1 {
+  @mixin eds-theme-typography-heading-1;
+}
+.text--h2 {
+  @mixin eds-theme-typography-heading-2;
+}
+.text--h3 {
+  @mixin eds-theme-typography-heading-3;
+}
+.text--h4 {
+  @mixin eds-theme-typography-heading-4;
+}
+.text--h5 {
+  @mixin eds-theme-typography-heading-5;
+}
+.text--sm {
+  @mixin eds-theme-typography-body-text-sm;
+}
+.text--xs {
+  @mixin eds-theme-typography-body-text-xs;
+}
+.text--caption {
+  @mixin eds-theme-typography-caption;
+}
+.text--overline {
+  @mixin eds-theme-typography-overline;
+}
+
+/**
+ * Color
+ */
+.text--alert {
+  color: var(--eds-color-other-red-800);
+}
+.text--base {
+  color: var(--eds-theme-color-body-foreground);
+}
+.text--brand {
+  color: var(--eds-color-brand-blue-700);
+}
+.text--info {
+  color: var(--eds-color-other-blue-700);
+}
+.text--inherit {
+  color: inherit;
+}
+.text--neutral {
+  color: var(--eds-color-neutral-500);
+}
+.text--success {
+  color: var(--eds-color-other-green-700);
+}
+.text--warning {
+  color: var(--eds-color-other-orange-700);
+}
+.text--white {
+  color: var(--eds-color-neutral-white);
+}
+
+/**
+ * Spacing
+ */
+.text--mb-half {
+  margin-bottom: var(--eds-size-half);
+}
+.text--mb-1x {
+  margin-bottom: var(--eds-size-1);
+}
+.text--mb-2x {
+  margin-bottom: var(--eds-size-2);
+}
+
+/**
+ * Weights
+ */
+.text--weight-normal {
```

Comment:
Not sure if appropriate bem syntax for spacing and weights. Added the "mb-" and "weight-" to distinguish nonspecific descriptors such as `-normal`

Reply:
Makes sense to me. Using the example `button--full-width` from the docs, I could also see the prop name and value reversed (ie `text--normal-weight`), but I dunno if we wanna be that persnickety.

Reply:
Will removing `spacing` as it can be applied with utility classes and as for weight, will reverse as per snickets